EssentialFeed/EssentialFeedTests/Image Comments/Image Comments API/LoadImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  LoadImageCommentsFromRemoteUseCaseTests.swift
+//  EssentialFeedTests
+//
+//  Created by Lukas Bahrle Santana on 19/01/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class LoadImageCommentsFromRemoteUseCaseTests:XCTestCase {
+	func test_init_doesNotRequestDataFromURL() {
+		let (_, client) = makeSUT()
+		
+		XCTAssertTrue(client.requestedURLs.isEmpty)
+	}
+	
+	func test_load_requestsDataFromURL() {
+		let url = URL(string: "https://comments-url.com")!
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load{ _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_loadTwice_requestsDataFromURLTwice() {
+		let url = URL(string: "https://comments-url.com")!
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load{ _ in }
+		sut.load{ _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_load_deliversErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		
+		expect(sut, toCompleteWith: failure(.connectivity)) {
+			client.complete(with: anyNSError())
+		}
+	}
+	
+	func test_load_deliversErrorOnNon2xxHTTPResponse() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [199, 201, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData)) {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
+			}
+		}
+	}
```

Comment:
This test setup is wrong - the test name says "non 2xx" but it uses `201` as a sample.

This test should fail if using `201` because that's a valid status code.

It's passing because it's using `anyData()` which cannot be decoded.

It should use `makeItemsJSON([])` instead of `anyData()`. Moreover, it shouldn't contain the `201` sample because that's a valid status code in this endpoint.

```suggestion
	func test_load_deliversErrorOnNon2xxHTTPResponse() {
		let (sut, client) = makeSUT()
		
		let samples = [199, 300, 400, 500]
		
		samples.enumerated().forEach { index, code in
			expect(sut, toCompleteWith: failure(.invalidData)) {
				client.complete(withStatusCode: code, data: makeItemsJSON([]), at: index)
			}
		}
	}
```